ui/contexts/gasFee.js

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { useGasFeeInputs } from '../hooks/gasFeeInput/useGasFeeInputs';
+
+export const GasFeeContext = createContext({});
+
+export const GasFeeContextProvider = ({
+  children,
+  defaultEstimateToUse,
+  transaction,
+  minimumGasLimit,
+  editGasMode,
+}) => {
+  const gasFeeDetails = useGasFeeInputs(
+    defaultEstimateToUse,
+    transaction,
+    minimumGasLimit,
+    editGasMode,
+  );
+  return (
+    <GasFeeContext.Provider value={gasFeeDetails}>
+      {children}
+    </GasFeeContext.Provider>
+  );
+};
+
+export function useGasFeeContext() {
+  return useContext(GasFeeContext);
+}
+
+GasFeeContextProvider.propTypes = {
+  children: PropTypes.node,
```

Comment:
Let's not forget to add `.isRequired` on things that should always be passed. This information is super helpful to know how to write and update unit tests in the absence of TypeScript because you know which props to pass into the component in the test (and which once you can opt to leave out).